internal/manifests/collector/daemonset.go

```diff
@@ -51,7 +51,7 @@ func DaemonSet(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelem
 				Spec: corev1.PodSpec{
 					ServiceAccountName: ServiceAccountName(otelcol),
 					InitContainers:     otelcol.Spec.InitContainers,
-					Containers:         []corev1.Container{Container(cfg, logger, otelcol, true)},
+					Containers:         append(otelcol.Spec.AdditionalContainers, Container(cfg, logger, otelcol, true)),
```

Comment:
Does the order matter? 

Should be the collector container first?

Reply:
The only time container order matters is if any container sets a lifecycle/postStart setting. That will block containers in the list after the postStart until its been executed. 

Outside of that all containers are started concurrently and ordering does not matter. Generally container inter-dependencies should to be handled by sidecars. To give a real example, with istio sidecar injection it, it inserts its container image first but it can take longer to start than the app container.